src/internal/kopia/backup_bases.go

```diff
@@ -381,7 +432,41 @@ func (bb *backupBases) fixupAndVerify(ctx context.Context) {
 		assistToKeep = append(assistToKeep, man)
 	}
 
+	// Now do the same for assist backups. Discard any that have overlapping
+	// reasons or if missing details or snapshot IDs.
+	assistBackupDrops := findNonUniqueAssistBackups(ctx, bb.assistBackups)
+
+	for _, bup := range bb.assistBackups {
+		if _, ok := assistBackupDrops[bup.ID]; ok {
+			continue
+		}
+
+		// Sanity checks on assist backups
+		deetsID := bup.StreamStoreID
+		if len(deetsID) == 0 {
+			deetsID = bup.DetailsID
+		}
+
+		if len(bup.SnapshotID) == 0 || len(deetsID) == 0 {
+			assistBackupDrops[bup.ID] = struct{}{}
+
+			logger.Ctx(ctx).Info(
+				"dropping assist backup, invalid snapshot or details ID",
+				"backup_id", bup.ID)
+
+			continue
+		}
+
+		assistBackupsToKeep = append(assistBackupsToKeep, bup)
+	}
+
+	// TODO(pandeyabs): Go through all reasons & ensure that the assist backups
```

Comment:
@ashmrtn Could you please comment on this? :) Should I get this TODO fixed before we merge this PR?

Reply:
huh good callout, I didn't think to check that when I wrote this function. I would say it would be nice to have, but not critical to getting this feature enabled. There's already a fair amount of testing for the base selection and merge code that should hopefully ensure that doesn't happen. We can pick this up when we have some spare time